velox/functions/sparksql/MightContain.cpp

```diff
@@ -33,26 +49,24 @@ class BloomFilterMightContainFunction final : public exec::VectorFunction {
     context.ensureWritable(rows, BOOLEAN(), resultRef);
     auto& result = *resultRef->as<FlatVector<bool>>();
     exec::DecodedArgs decodedArgs(rows, args, context);
-    auto serialized = decodedArgs.at(0);
     auto value = decodedArgs.at(1);
 
-    HashStringAllocator allocator{context.pool()};
-    VELOX_USER_CHECK(serialized->isConstantMapping())
-    BloomFilter output{StlAllocator<uint64_t>(&allocator)};
-    try {
-      auto sv = serialized->valueAt<StringView>(0);
-      output.merge(sv.data());
-    } catch (const std::exception& e) {
-      context.setErrors(rows, std::current_exception());
-      return;
+    if (!isSet_) {
+      rows.applyToSelected([&](int row) { result.set(row, false); });
+    } else {
+      rows.applyToSelected([&](int row) {
+        auto contain = bloom_.mayContain(
+            folly::hasher<int64_t>()(value->valueAt<int64_t>(row)));
+        result.set(row, contain);
+      });
     }
-
-    rows.applyToSelected([&](int row) {
-      auto contain = output.mayContain(
-          folly::hasher<int64_t>()(value->valueAt<int64_t>(row)));
-      result.set(row, contain);
-    });
   }
+
+ private:
+  std::shared_ptr<memory::MemoryPool> pool_;
+  HashStringAllocator allocator_;
```

Comment:
We don't need to reserve allocator_, just bloomFilter_ is enough

Reply:
@mbasmanova Per discussion with Chengcheng, we use `HashStringAllocator` instead of the default allocator here is because we want to track the memory usage of serialized Bloom Filter in context memory pool. But if the bloom filter is initialized only once per thread, it should be a small piece of memory. I have tested that for 4k and 8k batch size, the `bits_.size()` in BloomFilter are both 128KB so it might be unnecessary to track using the context memory pool. Do you think we can remove the pool_ and allocator_ here and create the BloomFilter with default allocator?

Reply:
@marin-ma This sounds reasonable to me.